### ISSUE TITLE
Sort using display value instead of to value of state_map

### ISSUE THREAD

**Issue by Gonioul** [maxwroc/battery-state-card#668]
**Describe the bug**
Sort is using display value instead of to value of state_map

Is this wanted?

**YAML configuration**
```yaml
              - type: custom:battery-state-card
                title: Pollens details
                sort:
                  by: state
                  desc: true
                collapse: 6
                tap_action: more-info
                icon: mdi:alert-decagram-outline
                state_map:
                  - from: "nul"
                    to: 0
                    display: "nul"
                  - from: "faible"
                    to: 1
                    display: "faible"
                  - from: "moyen"
                    to: 2
                    display: "moyen"
                  - from: "élevé"
                    to: 3
                    display: "élevé"
                colors:
                  steps:
                    - value: 0
                      color: var(--label-badge-grey)
                    - value: 1
                      color: var(--label-badge-green)
                    - value: 2
                      color: var(--label-badge-yellow)
                    - value: 3
                      color: var(--label-badge-red)
                filter:
                  include:
                    - name: entity_id
                      operator: matches
                      value: /^sensor.pollens_75_.*$/
                  exclude:
                    - name: entity_id
                      operator: matches
                      value: /^.*_risklevel$/
```

**Dev console errors**

**Screenshots**
![image](https://github.com/maxwroc/battery-state-card/assets/64291573/92779821-4643-48d9-8039-eb1dd0b6ce9b)

**Version**
v3.1.5

**Comment by maxwroc** [comment]
Yes this is true and kind of "by design" as of now. TBH I was thinking about changing this although I wanted to figure out a way which would not be confusing for people.

I think I will introduce a separate/special sorting option like `state_numeric` e.g.
```yaml
sort:
  by: state_numeric
```

Although the numeric state value may not exist (e.g. state_map is not defined in config) so it would need to default to the "state" / display value (this can be confusing)

**Comment by maxwroc** [comment]
Fixed in 3.2.0
